### ISSUE TITLE
Trusty ESM index should only be installed on i386/amd64

### ISSUE THREAD

**Issue by powersj** [CanonicalLtd/ubuntu-advantage-client#911]
Upstream bug: https://bugs.launchpad.net/ubuntu/+source/ubuntu-advantage-tools/+bug/1851858

**Comment by panlinux** [comment]
We can fix part of this in postinst, but we also need a fix for `ua attach`, as it tries to enable `esm-infra` again:
```
ubuntu@trusty-ppc64el:~$ sudo ua attach lolsecret
Enabling default service esm-infra
Updating package lists
Updating package lists
Failed to enable default services, check: sudo ua status
```
ESM remains disabled and with no advertisement, which is good since `apt-get update` keeps working.
The postinst fix I'm testing is:
```diff
--- a/debian/postinst
+++ b/debian/postinst
@@ -17,6 +17,8 @@ ESM_INFRA_APT_SOURCE_FILE_TRUSTY="$APT_SRC_DIR/ubuntu-esm-infra-trusty.list"
 ESM_APT_PREF_FILE_TRUSTY="/etc/apt/preferences.d/ubuntu-esm-trusty"
 ESM_INFRA_APT_PREF_FILE_TRUSTY="/etc/apt/preferences.d/ubuntu-esm-infra-trusty"
 
+MYARCH="$(dpkg --print-architecture)"
+ESM_SUPPORTED_ARCHS="i386 amd64"
 
 unconfigure_esm() {
     rm -f $APT_TRUSTED_KEY_DIR/ubuntu-esm*gpg  # Remove previous esm keys
@@ -78,8 +80,15 @@ case "$1" in
       rm -rf /var/cache/ubuntu-advantage-tools
 
       if [ "14.04" = "$VERSION_ID" ]; then
-        configure_esm
+        if echo "$ESM_SUPPORTED_ARCHS" | grep -qw "$MYARCH"; then
+          # 14.04 and supported arch
+          configure_esm
+        else
+          # 14.04 and unsupported arch
+          unconfigure_esm
+        fi
       else
+        # not 14.04, regardless of arch
         unconfigure_esm
       fi
       if [ ! -f /var/log/ubuntu-advantage.log ]; then

**Comment by panlinux** [comment]
Doesn't look like anything else needs to be done with the client. I asked the server devs to change the config on staging to allow only for i386/amd64 for esm-infra, and a quick test on ppc64el showed this working already, as the client didn't try to enable esm-infra on attach, nor was it advertised as being available on unattached status.

**Comment by panlinux** [comment]
Server-side PR, only in staging for now: https://github.com/CanonicalLtd/ua-contracts/pull/536

**Comment by panlinux** [comment]
There is this annoying discrepancy to sort out:
```
ubuntu@trusty-i386-ua:~$ dpkg --print-architecture 
i386

ubuntu@trusty-i386-ua:~$ sudo ua enable esm-infra
One moment, checking your subscription first
ESM Infra is not available for platform i686.
Supported platforms are: i386, x86_64
```

**Comment by panlinux** [comment]
Note it also differs for the x86_64 case (amd64)

**Comment by panlinux** [comment]
I filed https://github.com/CanonicalLtd/ubuntu-advantage-client/issues/914 for the arch madness.

**Comment by panlinux** [comment]
The contract server will start providing the architecture names from both `dpkg --print-architecture` and `uname -m`, see https://github.com/CanonicalLtd/ua-contracts/issues/539

**Comment by powersj** [comment]
Closing as fixed with #915.
